server/neptune/temporalworker/server.go

```diff
@@ -0,0 +1,192 @@
+package temporalworker
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/runatlantis/atlantis/server"
+	"github.com/runatlantis/atlantis/server/controllers"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/uber-go/tally"
+	"github.com/urfave/cli"
+	"github.com/urfave/negroni"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const (
+	AtlantisNamespace        = "atlantis"
+	DeployTaskqueue          = "deploy"
+	ProjectJobsViewRouteName = "project-jobs-detail"
+)
+
+// Config is TemporalWorker specific user config
+type Config struct {
+	CtxLogger        logging.Logger
+	DataDir          string
+	SslCertFile      string
+	SslKeyFile       string
+	TemporalHostPort string
+	Scope            tally.Scope
+	Closer           io.Closer
+}
+
+type httpServerProxy struct {
+	*http.Server
+	SSLCertFile string
+	SSLKeyFile  string
+	Logger      logging.Logger
+}
+
+func (p *httpServerProxy) ListenAndServe() {
+	var err error
+	if p.SSLCertFile != "" && p.SSLKeyFile != "" {
+		err = p.Server.ListenAndServeTLS(p.SSLCertFile, p.SSLKeyFile)
+	} else {
+		err = p.Server.ListenAndServe()
+	}
+	if err != nil && err != http.ErrServerClosed {
+		p.Logger.Error(err.Error())
+	}
+}
```

Comment:
I'd put this in like an http package.  so you can reference it like `http.ServerProxy` and we can then use it for gateway too.